store/sqlstore/channel_store.go

```diff
@@ -4337,48 +4337,45 @@ func (s SqlChannelStore) GetTopInactiveChannelsForTeamSince(teamID string, userI
 			LastActivityAt
 		FROM
 			((SELECT
-				Posts.ChannelId AS ID,
+				PublicChannels.Id AS ID,
 				'O' AS Type,
 				PublicChannels.DisplayName AS DisplayName,
 				PublicChannels.Name AS Name,
-				count(Posts.Id) AS MessageCount,
-				max(Posts.CreateAt) AS LastActivityAt
+				COALESCE(count(Posts.Id), 0) AS MessageCount,
+				COALESCE(max(Posts.CreateAt), 0) AS LastActivityAt
 			FROM
-				Posts
-				LEFT JOIN PublicChannels on Posts.ChannelId = PublicChannels.Id
+				PublicChannels
+				LEFT JOIN Posts on Posts.ChannelId = PublicChannels.Id AND Posts.Type = '' AND Posts.CreateAt > ? AND Posts.DeleteAt = 0
+				LEFT JOIN Channels on Channels.Id = PublicChannels.Id
```

Comment:
What's the reason for joining `Channels` into this subquery that already has `PublicChannels`?

Reply:
Channels.CreateAt is used to check if the channel was created before a particular time. This field is absent in `PublicChannels`